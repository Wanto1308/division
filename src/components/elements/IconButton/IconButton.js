import React from 'react';
import PropTypes from 'prop-types';
import { autoPx } from '../../../utils/unit';
import styles from './styles.scoped.css';

export default function IconButton(props) {
  const { className, children, disabled, icon, onClick, size } = props;
  const classes = [styles.root, className].filter(Boolean).join(' ');
  const newSize = autoPx(size);
  const style = { height: newSize, width: newSize };

  return (
    <button className={classes} disabled={disabled} onClick={onClick} style={style} type="button">
      {icon ? <img alt={icon} src={`/assets/ic-${icon}.svg`} /> : children}
    </button>
  );
}

IconButton.defaultProps = {
  children: null,
  className: '',
  disabled: false,
  icon: '',
  onClick: () => { },
  size: 24,
};

IconButton.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  size: PropTypes.number,
};
