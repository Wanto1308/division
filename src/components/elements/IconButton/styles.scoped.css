.root {
  background: none;
  border: none;
  border-radius: 0.125rem;
  cursor: pointer;
  transition: background-color 300ms;
}

.root:hover {
  background-color: var(--black-color-6);
}

.root:disabled {
  filter: opacity(0.4);
  pointer-events: none;
}

.root > img {
  font-size: xx-small;
  height: 100%;
  margin: auto;
}
