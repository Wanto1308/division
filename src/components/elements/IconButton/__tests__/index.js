import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import IconButton from '../IconButton';

jest.mock('../../../../utils/unit', () => ({
  autoPx: v => v
}));

describe('src/components/elements/IconButton', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<IconButton />);
    expect(tree).toMatchSnapshot();
    IconButton.defaultProps.onClick();
  });

  test('icon', () => {
    const result = IconButton({ ...IconButton.defaultProps, icon: 'test' });
    expect(result.props.children.type).toBe('img');
  });
});
