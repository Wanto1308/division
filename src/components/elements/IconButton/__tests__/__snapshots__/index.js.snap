// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/IconButton render 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "height": 24,
      "width": 24,
    }
  }
  type="button"
/>
`;
